modules/scylladb/scylladb.go

```diff
@@ -0,0 +1,214 @@
+		})
+
+		return nil
+	}
+}
+
+// WithCustomCommands sets custom commands with values for the ScyllaDB container.
+// Pass the command and the values as a list of strings in the following format: "--flag1=value", "--flag2", etc.
+// In case of an invalid flag (not starting with "--" or "-"), this option returns an error,
+// not applying any changes to the command line. Else, flags that exist in the command line overwrite the default commands.
+// See more in the [ScyllaDB docs].
+//
+// [ScyllaDB docs]: https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/tips/best-practices-scylla-on-docker.html
+func WithCustomCommands(flags ...string) testcontainers.CustomizeRequestOption {
+	var errInvalidFlag error
+
+	flagsMap := make(map[string]string)
+	for _, flag := range flags {
+		if !strings.HasPrefix(flag, "--") && !strings.HasPrefix(flag, "-") {
+			errInvalidFlag = fmt.Errorf("invalid flag: %s", flag)
+			break
+		}
+
+		before, after, found := strings.Cut(flag, "=")
+		if found {
+			flagsMap[before] = after
+		} else {
+			flagsMap[flag] = ""
+		}
+	}
+
+	return func(req *testcontainers.GenericContainerRequest) error {
+		if errInvalidFlag != nil {
+			return errInvalidFlag
+		}
+
+		setCommandFlag(req, flagsMap)
+		return nil
+	}
+}
+
+// ShardAwareConnectionHost returns the host and port of the ScyllaDB container with the shard-aware port
+func (c Container) ShardAwareConnectionHost(ctx context.Context) (string, error) {
+	return c.ConnectionHost(ctx, 19042)
+}
+
+// NonShardAwareConnectionHost returns the host and port of the ScyllaDB container with the non-shard-aware port
+func (c Container) NonShardAwareConnectionHost(ctx context.Context) (string, error) {
+	return c.ConnectionHost(ctx, 9042)
+}
+
+// ConnectionHost returns the host and port of the Scylladb container with the given port
+// and obtaining the host and exposed port from the container
+// Eg: "host:port" -> "localhost:9042" -> "localhost:19042" -> "localhost:8000"
+func (c Container) ConnectionHost(ctx context.Context, port uint16) (string, error) {
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", fmt.Errorf("host: %w", err)
+	}
+
+	natPort, err := nat.NewPort("tcp", strconv.FormatInt(int64(port), 10))
+	if err != nil {
+		return "", fmt.Errorf("new port: %w", err)
+	}
+
+	containerPort, err := c.MappedPort(ctx, natPort)
+	if err != nil {
+		return "", fmt.Errorf("mapped port: %w", err)
+	}
+
+	return net.JoinHostPort(host, containerPort.Port()), nil
+}
+
+// Run starts a ScyllaDB container with the specified image and options
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*Container, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{port},
+		Cmd: []string{
+			"--developer-mode=1",
+			"--overprovisioned=1",
+			"--smp=1",
+			"--memory=512M",
+		},
+		WaitingFor: wait.ForAll(
+			wait.ForLog(".*initialization completed..*").AsRegexp(),
```

Comment:
question: should this be removed, as log checks are fragile and I would expect the other checks to do the job?

Reply:
I kept it because it was added by the author (devrel at scylladb), and it's also present in the Java impl.

I agree that new modules should not rely on the log if they provide a more robust strategy, which is the case. I'm removing it as you suggest